cli/src/cli_util.rs

```diff
@@ -1439,7 +1423,17 @@ Set which revision the branch points to with `jj branch set {branch_name} -r <RE
         let mut locked_ws = self.workspace.start_working_copy_mutation()?;
         let old_op_id = locked_ws.locked_wc().old_operation_id().clone();
         let (repo, wc_commit) =
-            match check_stale_working_copy(locked_ws.locked_wc(), &wc_commit, &repo)? {
+            match check_stale_working_copy(locked_ws.locked_wc(), &wc_commit, &repo).map_err(
+                |e| match e {
+                    OpStoreError::ObjectNotFound { .. } => user_error_with_hint(
+                        "Could not read working copy's operation.",
+                        "Run `jj workspace update-stale` to recover.
+See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-working-copy \
+                         for more information.",
+                    ),
+                    e => e.into(),
+                },
+            )? {
```

Comment:
nit: at your option, this can be organized as follows:
```rust
match check_stale_working_copy(locked_ws.locked_wc(), &wc_commit, &repo) {
    Ok(WorkingCopyFreshness::Fresh) => ...
    ...
    Err(OpStoreError::ObjectNotFound { .. }) => ...
    Err(err) => return Err(err.into()),
}
```

Reply:
Thanks, I've used what you suggested.